x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_form_error.tsx

```diff
@@ -4,29 +4,161 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
-import { FormattedMessage } from '@kbn/i18n/react';
-import { EuiSpacer, EuiCallOut } from '@elastic/eui';
+import React, { useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiSpacer, EuiCallOut, EuiFlexGroup, EuiFlexItem, EuiButtonEmpty } from '@elastic/eui';
+import { useKibana } from '../../../shared_imports';
 
 interface Props {
-  errorMessage: string;
+  error: unknown;
 }
+interface PipelineError {
+  reason: string;
+  processorType?: string;
+}
+interface PipelineErrors {
+  errors: PipelineError[];
+}
+
+interface ErrorNode {
+  reason: string;
+  processor_type?: string;
+  suppressed?: ErrorNode[];
+}
+
+interface ErrorAttributesObject {
+  attributes: {
+    error: {
+      root_cause: [ErrorNode];
+    };
+  };
+}
+
+const flattenErrorsTree = (node: ErrorNode): PipelineError[] => {
+  const result: PipelineError[] = [];
+  const recurse = (_node: ErrorNode) => {
+    result.push({ reason: _node.reason, processorType: _node.processor_type });
+    if (_node.suppressed && Array.isArray(_node.suppressed)) {
+      _node.suppressed.forEach(recurse);
+    }
+  };
+  recurse(node);
+  return result;
+};
+
+const toKnownError = (error: unknown): PipelineErrors => {
+  if (
+    typeof error === 'object' &&
+    error != null &&
+    (error as any).attributes?.error?.root_cause?.[0]
+  ) {
+    const errorAttributes = error as ErrorAttributesObject;
+    const rootCause = errorAttributes.attributes.error.root_cause[0];
+    return { errors: flattenErrorsTree(rootCause) };
+  }
+
+  if (typeof error === 'string') {
+    return { errors: [{ reason: error }] };
+  }
+
+  if (
+    error instanceof Error ||
+    (typeof error === 'object' && error != null && (error as any).message)
+  ) {
+    return { errors: [{ reason: (error as any).message }] };
+  }
+
+  return { errors: [{ reason: 'An unknown error occurred.' }] };
+};
+
+const title = i18n.translate('xpack.ingestPipelines.form.savePipelineError', {
+  defaultMessage: 'Unable to create pipeline',
+});
+
+export const PipelineFormError: React.FunctionComponent<Props> = ({ error }) => {
+  const { services } = useKibana();
+  const [isShowingAllErrors, setIsShowingAllErrors] = useState<boolean>(false);
+  const safeErrorResult = toKnownError(error);
+  const hasMoreErrors = safeErrorResult.errors.length > 5;
```

Comment:
What do you think about extracting `5` as a const, so it's clear it represents the initial number of errors that will be shown?